driver-sync/src/test/functional/com/mongodb/client/OidcAuthenticationProseTests.java

```diff
@@ -0,0 +1,937 @@
+        }
+    }
+
+    private void performFind(final MongoClient mongoClient) {
+        mongoClient
+                .getDatabase("test")
+                .getCollection("test")
+                .find()
+                .first();
+    }
+
+    private static <T extends Throwable> void assertCause(
+            final Class<T> expectedCause, final String expectedMessageFragment, final Executable e) {
+        Throwable actualException = assertThrows(Throwable.class, e);
+        assertCause(expectedCause, expectedMessageFragment, actualException);
+    }
+
+    private static <T extends Throwable> void assertCause(
+            final Class<T> expectedCause, final String expectedMessageFragment, final Throwable actualException) {
+        Throwable cause = actualException;
+        while (cause.getCause() != null) {
+            cause = cause.getCause();
+        }
+        if (!expectedCause.isInstance(cause)) {
+            throw new AssertionFailedError("Unexpected cause", actualException);
+        }
+        if (!cause.getMessage().contains(expectedMessageFragment)) {
+            throw new AssertionFailedError("Unexpected message", actualException);
+        }
+    }
+
+    protected void delayNextFind() {
+        try (MongoClient client = createMongoClient(createSettings(AWS_OIDC_URL, null, null))) {
+            BsonDocument failPointDocument = new BsonDocument("configureFailPoint", new BsonString("failCommand"))
+                    .append("mode", new BsonDocument("times", new BsonInt32(1)))
+                    .append("data", new BsonDocument()
+                            .append("appName", new BsonString(appName))
+                            .append("failCommands", new BsonArray(asList(new BsonString("find"))))
+                            .append("blockConnection", new BsonBoolean(true))
+                            .append("blockTimeMS", new BsonInt32(100)));
+            client.getDatabase("admin").runCommand(failPointDocument);
+        }
+    }
+
+    protected void failCommand(final int code, final int times, final String... commands) {
+        try (MongoClient mongoClient = createMongoClient(createSettings(
+                AWS_OIDC_URL, null, null))) {
+            List<BsonString> list = Arrays.stream(commands).map(c -> new BsonString(c)).collect(Collectors.toList());
+            BsonDocument failPointDocument = new BsonDocument("configureFailPoint", new BsonString("failCommand"))
+                    .append("mode", new BsonDocument("times", new BsonInt32(times)))
+                    .append("data", new BsonDocument()
+                            .append("appName", new BsonString(appName))
+                            .append("failCommands", new BsonArray(list))
+                            .append("errorCode", new BsonInt32(code)));
+            mongoClient.getDatabase("admin").runCommand(failPointDocument);
+        }
+    }
```

Comment:
[optional]

One may use something like

```java
try (FailPoint autoRemoved = FailPoint.enable(configureFailPoint, createSettings(AWS_OIDC_URL, null, null).getClusterSettings().getHosts().get(0))) {
    ...
}
```
to automatically remove failpoints. Though `com.mongodb.client.FailPoint.close` can't guarantee that the failpoint is removed, as it requires communicating with the server.

Reply:
I had a look but did not see that settings could be passed in, and thought making certain changes might make the API uglier. I also tried creating a "clear failpoints" method that set a failpoint and then cleared it - I thought I previously observed that only one failpoint could exist at one time. But that did not work. So I used a varying appName to accomplish the same, as the prose tests suggested.